pages/api/auth/agent-connect/callback.ts

```diff
@@ -10,6 +14,26 @@ export default withSession(async function callbackRoute(req, res) {
   try {
     const userInfo = await agentConnectAuthenticate(req);
     const agent = await getAgent(userInfo);
+
+    const isWhitelisted = agent.scopes.some((scope) => isAgentScope(scope));
+    const { isMCP } = agent;
+
+    if (!isWhitelisted && isMCP) {
+      const siren = extractSirenFromSiret(agent.siret);
+      const uniteLegale = await clientUniteLegaleRechercheEntreprise(siren, 0);
+
```

Comment:
I would rather use `getUniteLegaleFromSlug` here as we want redundancy

Reply:
It's not possible to use getUniteLegaleFromSlug because it uses next/headers.
What do you advise ?
Using Client OR migrating to app/api OR adapt getUniteLegaleFromSlug ?